### PR TITLE
Add mqtt retained information to pubsub message metadata

### PR DESCRIPTION
**Pull request by thielepaul**

### CODE REVIEW

**Comment by berndverst** [comment]
@thielepaul please sign-off on your commit.

https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work

https://github.com/dapr/components-contrib/pull/2299/checks?check_run_id=9616824234

File: pubsub/mqtt/mqtt.go [hunk 1]
```diff
@@ -281,8 +281,9 @@ func (m *mqttPubSub) onMessage(ctx context.Context) func(client mqtt.Client, mqt
 		}()
 
 		msg := pubsub.NewMessage{
-			Topic: mqttMsg.Topic(),
-			Data:  mqttMsg.Payload(),
+			Topic:    mqttMsg.Topic(),
+			Data:     mqttMsg.Payload(),
+			Metadata: map[string]string{"retained": fmt.Sprint(mqttMsg.Retained())},
```
Comment (ItalyPaleAle) [inline pubsub/mqtt/mqtt.go:286]: Can you use strconv to convert to string? Otherwise, LGTM

Comment (thielepaul) [reply pubsub/mqtt/mqtt.go]: thanks for the suggestion! I changed it to strconv

**Review by ItalyPaleAle — APPROVED** [review]

**Review by pravinpushkar — APPROVED** [review]

**Review by ItalyPaleAle — APPROVED** [review]

**Comment by thielepaul** [comment]
Is there something blocking this PR from being merged? If yes, please let me know :)

**Comment by berndverst** [comment]
> Is there something blocking this PR from being merged? If yes, please let me know :)

Yes, it's called the holidays. The holidays were blocking this from being merged. Now that we are back to work we will look into it. Thanks for your patience.

**Review by berndverst — APPROVED** [review]

**Pull request merged**